src/common/telemetry-data-factory.ts

```diff
@@ -523,4 +538,43 @@ export class TelemetryDataFactory {
             enabled,
         };
     }
+
+    public forCompletedTestStep(
```

Comment:
Why are we moving this to the telemetry data factory? This seems very relevant to the "completed-test-step-telemetry" work where it constructs that telemetry data and the telemetry-data-factory can be responsible for putting it together.

Reply:
Agreed